File: javascript/test/node-test.js
const assert = require('chai').assert;

const createNode = require('../lib/node');

describe('createNode', () => {

  it('returns an object', () => {
    assert.typeOf(createNode('some data'), 'object');
  });

  it('creates a node with a data property', () => {
    assert.property(createNode('some data'), 'data');
  });

  it('creates a node with a next property', () => {
    assert.property(createNode('some data'), 'next');
  });

});

File: ruby/test/node_test.rb
require "minitest/autorun"
require "minitest/spec"
require "minitest/pride"

require "./lib/node"

describe Node do

  it "responds to 'data'" do
    node = Node.new("some data")

    assert_respond_to(node, :data)
  end

  it "responds to 'next'" do
    node = Node.new("some data")

    assert_respond_to(node, :next)
  end

  it "had the data passed in" do
    node = Node.new("some data")

    assert_equal(node.data, "some data")
  end

end